http-client/src/main/java/com/tutrit/httpclient/gateway/CustomerGateway/HttpCustomerGateway.java

```diff
@@ -0,0 +1,93 @@
+package com.tutrit.httpclient.gateway.CustomerGateway;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tutrit.bean.Customer;
+import com.tutrit.gateway.CustomerGateway;
+import com.tutrit.httpclient.gateway.config.EndpointConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+
+@Qualifier
+@Component
+public class HttpCustomerGateway implements CustomerGateway {
+    @Autowired
+    private EndpointConfig endpointConfig;
+
+
+    private static final HttpClient httpClient = HttpClient.newBuilder()
+            .version(HttpClient.Version.HTTP_2)
+            .build();
+
+    @Override
+    public Customer saveCustomer(Customer customer) {
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(endpointConfig.getRestApiUrl()))
+                .POST(HttpRequest.BodyPublishers
+                        .ofString(String.format(
+                                "{\"customerId\":\"%s\"\"name\":\"%s\",\"city\":\"%s\",\"phoneNumber\":\"%s\",\"email\":\"%s\"}",
+                                customer.customerId(), customer.name(), customer.city(), customer.phoneNumber(), customer.email())))
```

Comment:
this should be posted as body, maybe map from string ? 
good example here https://stackoverflow.com/questions/66388231/json-object-to-body-publisher